core/variables.js

```diff
@@ -75,6 +76,22 @@ Blockly.Variables.allVariables = function(root) {
 };
 
 /**
+ * Find all variables that the user has created through the workspace or
+ * toolbox.  For use by generators.
+ * @param {!Blockly.Workspace} root The workspace to inspect.
+ * @return {!Array.<string>} Array of variable names.
+ */
+Blockly.Variables.allVariables = function(root) {
+  if (root.getDescendants) {
```

Comment:
46/47 is less about determining type, and more about verifying the existence of a function before calling it.  That could be done either way, but you are right, testing with instanceof would be a bit better.